extension/textext/asktext.py

```diff
@@ -483,11 +427,62 @@ def select_preamble_file(self):
                 self._preamble.delete(0, Tk.END)
                 self._preamble.insert(Tk.END, file_name)
 
+        def error_dialog(self, title, label, error):
+            """
+            Present an error dialog
+
+            :param parent: Parent window
+            :param title: Error title text
+            :param label: Label text
+            :param error: exception
+            :type error: StandardError
+            """
+
+            def error_dialog_close():
+                # ToDo: Check Windows behavior!! -disable
+                self._root.wm_attributes("-topmost", True)
+                self._err_dialog.destroy()
+
+            def add_textview(header, text):
+                err_dialog_frame = Tk.Frame(self._err_dialog)
+                err_dialog_label = Tk.Label(err_dialog_frame, text=header)
+                err_dialog_label.pack(side='top', fill=Tk.X)
+                err_dialog_text = Tk.Text(err_dialog_frame, height=10)
+                err_dialog_text.insert(Tk.END, text)
+                err_dialog_text.pack(side='left', fill=Tk.Y)
+                err_dialog_scrollbar = Tk.Scrollbar(err_dialog_frame)
+                err_dialog_scrollbar.pack(side='right', fill=Tk.Y)
+                err_dialog_scrollbar.config(command=err_dialog_text.yview)
+                err_dialog_text.config(yscrollcommand=err_dialog_scrollbar.set)
+                err_dialog_frame.pack(side='top')
+
+            # ToDo: Check Windows behavior!! --> -disable
+            self._root.wm_attributes("-topmost", False)
+
+            self._err_dialog = Tk.Toplevel(self._frame)
+            self._err_dialog.minsize(300, 400)
+            self._err_dialog.transient(self._frame)
+            self._err_dialog.focus_force()
+            self._err_dialog.grab_set()
+            self._err_dialog.protocol("WM_DELETE_WINDOW", error_dialog_close)
```

Comment:
Here `error_dialog_close` can capture local `err_dialog` just being moved down by 30 lines. (also `close_error_dialog` is better name)

Try to not store temporary variables as `self` hidden `attributes` whenever possible. It will reduce number of "states" of `self` and make reasoning about program easier.